Libraries/LibJS/Runtime/SymbolObject.cpp

```diff
@@ -0,0 +1,102 @@
+#include <LibJS/Runtime/Symbol.h>
+#include <LibJS/Runtime/SymbolObject.h>
+#include <LibJS/Runtime/SymbolPrototype.h>
+#include <LibJS/Runtime/Value.h>
+
+namespace JS {
+
+HashMap<String, Value> SymbolObject::s_global_symbol_map;
+
+Value SymbolObject::s_well_known_iterator;
+Value SymbolObject::s_well_known_async_terator;
+Value SymbolObject::s_well_known_match;
+Value SymbolObject::s_well_known_match_all;
+Value SymbolObject::s_well_known_replace;
+Value SymbolObject::s_well_known_search;
+Value SymbolObject::s_well_known_split;
+Value SymbolObject::s_well_known_has_instance;
+Value SymbolObject::s_well_known_is_concat_spreadable;
+Value SymbolObject::s_well_known_unscopables;
+Value SymbolObject::s_well_known_species;
+Value SymbolObject::s_well_known_to_primtive;
+Value SymbolObject::s_well_known_to_string_tag;
+
+SymbolObject* SymbolObject::create(GlobalObject& global_object, Symbol& primitive_symbol)
+{
+    return global_object.heap().allocate<SymbolObject>(primitive_symbol, *global_object.symbol_prototype());
+}
+
+SymbolObject::SymbolObject(Symbol& symbol, Object& prototype)
+    : Object(&prototype)
+    , m_symbol(symbol)
+{
+}
+
+SymbolObject::~SymbolObject()
+{
+}
+
+Value SymbolObject::get_global(Interpreter& interpreter, String description)
+{
+    auto global_symbol = s_global_symbol_map.get(description);
+    if (global_symbol.has_value())
+        return global_symbol.value();
+
+    auto symbol = js_symbol(interpreter, description, true);
+    s_global_symbol_map.set(description, symbol);
+    return Value(symbol);
+}
+
+void SymbolObject::initialize_well_known_symbols(Interpreter& interpreter)
+{
+    SymbolObject::s_well_known_iterator = Value(js_symbol(interpreter, "Symbol.iterator", false));
+    SymbolObject::s_well_known_async_terator = Value(js_symbol(interpreter, "Symbol.asyncIterator", false));
+    SymbolObject::s_well_known_match = Value(js_symbol(interpreter, "Symbol.match", false));
+    SymbolObject::s_well_known_match_all = Value(js_symbol(interpreter, "Symbol.matchAll", false));
+    SymbolObject::s_well_known_replace = Value(js_symbol(interpreter, "Symbol.replace", false));
+    SymbolObject::s_well_known_search = Value(js_symbol(interpreter, "Symbol.search", false));
+    SymbolObject::s_well_known_split = Value(js_symbol(interpreter, "Symbol.split", false));
+    SymbolObject::s_well_known_has_instance = Value(js_symbol(interpreter, "Symbol.hasInstance", false));
+    SymbolObject::s_well_known_is_concat_spreadable = Value(js_symbol(interpreter, "Symbol.isConcatSpreadable", false));
+    SymbolObject::s_well_known_unscopables = Value(js_symbol(interpreter, "Symbol.unscopables", false));
+    SymbolObject::s_well_known_species = Value(js_symbol(interpreter, "Symbol.species", false));
+    SymbolObject::s_well_known_to_primtive = Value(js_symbol(interpreter, "Symbol.toPrimitive", false));
+    SymbolObject::s_well_known_to_string_tag = Value(js_symbol(interpreter, "Symbol.toStringTag", false));
```

Comment:
Who protects these from garbage collection?

Reply:
Ahh, I didn't consider garbage collection. I assume that because they're static, I couldn't add them to a `visit_children` method. Would the best place to do that be `Interpreter.gather_roots`?

Reply:
Sure, perhaps we could add a public static function to `SymbolObject`, and then call that from `Interpreter::gather_roots()`. That way the values can still live in `SymbolObject.cpp` :)